pkg/kubelet/kubelet_node_status.go

```diff
@@ -469,15 +469,17 @@ func (kl *Kubelet) setNodeAddress(node *v1.Node) error {
 			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
 		}
 		if kl.nodeIP != nil {
+			enforcedNodeAddresses := []v1.NodeAddress{}
 			for _, nodeAddress := range nodeAddresses {
 				if nodeAddress.Address == kl.nodeIP.String() {
-					node.Status.Addresses = []v1.NodeAddress{
-						{Type: nodeAddress.Type, Address: nodeAddress.Address},
-						{Type: v1.NodeHostName, Address: kl.GetHostname()},
-					}
-					return nil
+					enforcedNodeAddresses = append(enforcedNodeAddresses, v1.NodeAddress{Type: nodeAddress.Type, Address: nodeAddress.Address})
 				}
 			}
+			if len(enforcedNodeAddresses) > 0 {
+				enforcedNodeAddresses = append(enforcedNodeAddresses, v1.NodeAddress{Type: v1.NodeHostName, Address: kl.GetHostname()})
```

Comment:
this is preexisting, but while we're here, nothing would prevent a cloud provider from returning the IP as a type Hostname as well... if we double-set the hostname type, the update will fail.

Reply:
Right, I didn't want to change this part, as this is how it used to work previously.
I can update the PR to only add a `NodeHostName` address if none is already defined or I can create a new PR specifically for this change, once this one is approved and applied.
@liggitt What do you prefer ?

Reply:
Separate PR is fine